tests/std/tests/GH_002168_regex_overflow/test.cpp

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <regex>
+
+using namespace std;
+
+bool shouldThrow(const char* const regexString) {
+    try {
+        regex regex{regexString, regex_constants::ECMAScript};

```

Comment:
Avoid calling a variable `regex`. There's an intention that "search all" on standard identifiers would show only these standard identifiers used as such; i.e. "regex" would only find `regex` as the class name.